agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4266,6 +4269,82 @@ bool slave_thread::send_cmdu_to_controller(ieee1905_1::CmduMessageTx &cmdu_tx)
                                   backhaul_params.bridge_mac);
 }
 
+/**
+ * @brief Diffie-Hellman public key exchange keys calculation
+ *        class member params authkey and keywrapauth are computed
+ *        on success.
+ *
+ * @param m1 WSC M1 TLV received from the radio agent
+ * @return true on success
+ * @return false on failure
+ */
+bool slave_thread::autoconfig_wsc_calculate_keys(std::shared_ptr<ieee1905_1::tlvWscM2> m2)
+{
+    auto mac = network_utils::mac_from_string(backhaul_params.bridge_mac);
+    if (!mapf::encryption::wps_calculate_keys(
+            dh, m2->public_key_attr().data, m2->public_key_attr().data_length,
+            m2->enrolee_nonce_attr().data, mac.oct, m2->registrar_nonce_attr().data, authkey,
+            keywrapkey)) {
+        LOG(ERROR) << "Failed to calculate WPS keys";
+        return false;
+    }
+
+    return true;
+}
+
+bool slave_thread::autoconfig_wsc_parse_m2_encrypted_settings(
+    std::shared_ptr<ieee1905_1::tlvWscM2> m2, std::string &ssid, std::string &bssid,
+    WSC::eWscAuth &auth_type, WSC::eWscEncr &encr_type)
+{
+    auto encrypted_settings = m2->encrypted_settings();
+    uint8_t *iv             = reinterpret_cast<uint8_t *>(encrypted_settings->iv());
```

Comment:
It is declared as char since the uint8_t doesn't have a convinient getter function and instead returns that weird tupple. Does it matter?

Reply:
> It is declared as char since the uint8_t doesn't have a convinient getter function and instead returns that weird tupple. Does it matter?

OK, I see you created a task for that.